### DISCUSSION TITLE
Question concerning the function input

### DISCUSSION THREAD

**Discussion by kadir-orhan** [pyTMD/pyTMD#79] [Q&A]
I am trying to run the function below to predict water levels on model grid points for generating an initial conditions file:
predict_tide(t, hc, constituents, DELTAT=0.0, CORRECTIONS='OTIS')

Although it is a straightforward function, I'm struggling to provide the input parameters in the right format, and I couldn't find a relevant example or explanation in the documentation. Can you please explain a bit more in detail how to define "**hc**" harmonic constant vector and "**constituents**" tidal constituent IDs?

**Edit:**
I have also tried "compute_tide_corrections" function, thinking it could be more suitable.

wl_ini[ii,jj] = tmd.compute_tide_corrections(model['X'][ii,jj], model['Y'][ii,jj], delta_time, \
                                                                      DIRECTORY = 'c:\\Users\\user01\\Scripts_Python', MODEL = 'TPXO9-atlas-v4', \
                                                                      EPSG = 4326, EPOCH = (2000,1,1,0,0,0), TYPE = 'grid', TIME = 'UTC', \
                                                                      METHOD = 'spline', EXTRAPOLATE = False, FILL_VALUE = np.nan)

This is the line I have tried to run. "**wl_ini[ii,jj]**" is the tidal water level I am trying to predict. "**model['X'][ii,jj]**" is a single longitude (106.55), "**model['Y'][ii,jj]**" is a single latitude (-5.45), **delta_time** is the seconds since EPOCH (675993600) and the rest of the arguments are defined based on the documentation. But somehow it doesn't work. The error I am getting is:
**cannot reshape array of size 175089971 into shape (1853423616,2)**

Do you have any idea what am I doing wrong?

**Comment by tsutterley** [comment]
Hi @kadir-orhan,
Great question.  You were doing absolutely everything right I just didn't have it in that function.
I just put in PR #80 that allows you to use `compute_tide_corrections` with `TYPE='time series'`.  This should handle the time series case.  Let me know if you have any problems!

**Reply by kadir-orhan** [reply]
Hi Tyler,

Thank you for the prompt answer during the holiday season :)

I think I have failed to make my point. I have tried to calculate tidal water levels at a certain location for only a single time point. I am just trying to define the initial conditions (at the first time step) at each grid node of a hydrodynamic model. I don't need time series. I have already extracted harmonic constituents using pyTMD to be used as boundary conditions later on :)

As I have tried to explain above, I gave the coordinates (x, y) as floats, delta_time as an integer (seconds since EPOCH), and DIRECTORY/MODEL for TPXO9-atlas-v4 was set as it was asked in the documentation, with the specific directory. I have opted for 'grid' for the 'TYPE' input parameter since that TPXO9-atlas-v4 provides gridded data. I thought something went wrong with one of these. Maybe I should have used list type input instead of floats/integers, as was the case for the 'model_files' input parameter for the 'extract_netcdf_constants' function?

If it comes down to the update you have just provided, I need to wait for the conda-forge to bring it in. I am trying to keep my packages consistent with conda archive, to prevent inconsistencies :) But again, thank you very much for your help, and I already wish you a happy new year.

**Reply by tsutterley** [reply]
Sorry @kadir-orhan I misunderstood looking at the line of code you put up earlier.  The `TYPE` argument describes the data you are submitting.  There's at present a limited set of `TYPE` options (`'grid'`, `'drift'`, and `'time series'`).  There is a function (`pyTMD.spatial.data_type`) that tries to figure it out based on the input dimensions of your spatial and temporal coordinates.

- `grid`: 2D spatial arrays with a common time point (such as an image)
- `drift`: arrays with each point being unique in time and space (such as a drift buoy)
- `time series`: single spatial points with multiple independent time points

I think `drift` or `time series` would work for a single point in time and space.  But if you're iterating over all of your model points this would be pretty slow (as there's no caching of the tide model so it'd have to be reread the tide files for every point).  I think the better option (if you need the tide values at more than one point) would be to use `grid` type with all the points of your model and then to mask it to the appropriate domain.

I'll put in a code release after finishing this message so that things are updated on conda-forge.  I've been thinking about it for a while :+1: 

Happy holidays to you as well.  Hope all is well your way.
